### PR TITLE
separate create to command-controller-service

### PR DESCRIPTION
**Pull request by davidmeirlevy**

### CODE REVIEW

**Review by itays123 — COMMENTED** [review]
looks good, I'll implement it in the yargs branch

File: controllers/upgrade.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const fs = require('fs');
+const { getJSON, checkAndUpgradeDependency } = require('../services/upgrade')
+const remotePackagePath = 'https://raw.githubusercontent.com/greenpress/greenpress/master/package.json';
+const { blue } = require('../utils/colors');
+
+async function upgradeController() {
+	const localPackagePath = process.env.PWD + '/package.json';
+
+	// create local json objects for package.json from local and remote repos
+	const localPackage = require(localPackagePath);
```
Comment (davidmeirlevy) [inline controllers/upgrade.js:10]: we should have a "getLocalPackage()" and "getRemoteDependencies()" at the service..

Comment (drorrose) [reply controllers/upgrade.js]: Done

File: controllers/upgrade.js [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
 const { blue } = require('../utils/colors');
 
 async function upgradeController() {
-	const localPackagePath = process.env.PWD + '/package.json';
-
 	// create local json objects for package.json from local and remote repos
-	const localPackage = require(localPackagePath);
-	const { dependencies = {} } = await getJSON(remotePackagePath);
+	const localPackage = getLocalPackage();
+	const { dependencies = {} } = await getJSON(getRemotePackage());
```
Comment (davidmeirlevy) [inline controllers/upgrade.js:9]: What??
getRemotePackage() should return the value, the “getJson” should be done from it..

Comment (drorrose) [reply controllers/upgrade.js]: Sorry, had the wrong idea about it. Fixed now.

**Pull request merged**